scripts/breaking_changes_checker/checkers/method_overloads_checker.py

```diff
@@ -0,0 +1,39 @@
+from _models import ChangesChecker
+import jsondiff
+
+class MethodOverloadsChecker(ChangesChecker):
+    name = "RemovedMethodOverload"
+    message = {
+        "default": "{}.{} had an overload `{}` removed",
+        "all": "{}.{} had all overloads removed"
+    }
+
+    def run_check(self, diff, stable_nodes, current_nodes):
+        bc_list = []
+        for module_name, module in diff.items():
+            for class_name, class_components in module.get("class_nodes", {}).items():
+                # We aren't checking for deleted classes in this checker
+                if isinstance(class_name, jsondiff.Symbol):
+                    continue
```

Comment:
I have a proposal for how we can drop folks down directly into the methods instead of having everyone have to loop through each module, class, etc. Adding it in this way until I get some feedback on the other design.